build/flags.mk

```diff
@@ -64,5 +64,11 @@ ALL_CPPFLAGS = $(TARGET_INCLUDES) $(INCLUDES) $(TARGET_CPPFLAGS) $(CPPFLAGS)
 ALL_CXXFLAGS = $(OPTIMIZE) $(TARGET_OPTIMIZE) $(FLAGS_PROFILE) $(SANITIZE_FLAGS) $(CXX_FEATURES) $(TARGET_CXXFLAGS) $(CXX_WARNINGS) $(CXXFLAGS)
 ALL_CFLAGS = $(OPTIMIZE) $(TARGET_OPTIMIZE) $(FLAGS_PROFILE) $(SANITIZE_FLAGS) $(C_FEATURES) $(C_WARNINGS) $(CFLAGS)
 
-ALL_LDFLAGS = $(OPTIMIZE_LDFLAGS) $(TARGET_LDFLAGS) $(FLAGS_PROFILE) $(SANITIZE_FLAGS) $(LDFLAGS)
+ifeq ($(LTO),y)
+  # Pass warning flags to the linker when using LTO optimizations. LTO means that second stage compilation happens here,
+  # and compiler warnings can occur here too.
+  ALL_LDFLAGS = $(OPTIMIZE_LDFLAGS) $(CXX_WARNINGS) $(TARGET_LDFLAGS) $(FLAGS_PROFILE) $(SANITIZE_FLAGS) $(LDFLAGS)
```

Comment:
This duplicates the whole line - can we do that without having two copies of this line?

Reply:
Use += instead of duplicating the lines.

Reply:
These `make` directives now behave differently than before (early variable expansion, because `+=` enforces it like `:=`, unlike `=` which is lazy). Did you check whether this is safe in all possible cases?

Reply:
`+=` on a lazy assigned variable preserves its lazyness in GNU make. The variables are not early expanded by the `+=`.
Proof: Write this into a Makefile:

```
BEFORE = Before

COMBINE = $(BEFORE)
COMBINE +=  $(AFTER)

AFTER = After

all:
	@echo $(COMBINE)
```

The output of `make` is "Before After". The lazy assigment of `After` to `AFTER` is preserved.

Reply:
Oh, you're right!